File: aat-android/src/main/java/ch/bailu/aat/services/dem/tile/DemProvider.java
package ch.bailu.aat.services.dem.tile;

public interface DemProvider {


    short getElevation(int index);
    DemDimension getDim();

    float getCellsize();

    boolean inverseLatitude();
    boolean inverseLongitude();


    DemProvider NULL = new DemProvider() {

        @Override
        public short getElevation(int index) {
            return 0;
        }

        @Override
        public DemDimension getDim() {
            return Dem3Array.DIMENSION;
        }

        @Override
        public float getCellsize() {
            return 50;
        }

        @Override
        public boolean inverseLatitude() {
            return false;
        }

        @Override
        public boolean inverseLongitude() {
            return false;
        }

    };
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/elevation/tile/DemProvider.kt
package ch.bailu.aat_lib.service.elevation.tile

interface DemProvider {
    /**
     * Elevation at a specific index in the elevation matrix
     */
    fun getElevation(index: Int): Short

    fun getDimension(): DemDimension

    /**
     * Distance in meter across a row/line
     */
    fun getCellDistance(): Float
    fun hasInverseLatitude(): Boolean
    fun hasInverseLongitude(): Boolean

    companion object {
        val NULL: DemProvider = object : DemProvider {
            override fun getElevation(index: Int): Short {
                return 0
            }

            override fun getDimension(): DemDimension {
                return Dem3Array.dem3Dimension
            }

            override fun getCellDistance(): Float {
                return 50f
            }

            override fun hasInverseLatitude(): Boolean {
                return false
            }

            override fun hasInverseLongitude(): Boolean {
                return false
            }
        }
    }
}
